### ISSUE TITLE
Server configuration with Matrix API

### ISSUE THREAD

**Issue by curiousjazz77** [CSims20/Eclipse-#1]
Hey Everyone,

Starting an issue since we need help starting it up. @mayureshjakhotia

**Comment by ThuyNT13** [comment]
just to recap, @CSims20 and @mayureshjakhotia are not coming in tomorrow. @curiousjazz77 will you be able to make it? if not, guess it will just be lil' ol' me.  ¯\_(ツ)_/¯ 

What needs to be worked on: 
- [ ] check out [Matrix Client-Server JS SDK](https://github.com/matrix-org/matrix-js-sdk), having abandoned Synapse
- [ ] Figure out how LAMP/MAMP works to integrate the user registration/login PHP/BCrypt code that Mayuresh was able to implement [PHP BCrypt](https://github.com/CSims20/Eclipse-/tree/master/phpbcrypt)

**Comment by curiousjazz77** [comment]
@ThuyNT13, I'm coming! I'll only be able to stay from 10-11:15/30

**Comment by mayureshjakhotia** [comment]
Hi Guys, I was working on the Matrix Client-Server API but was unable to run it. So debugging it and also seeing if there is some other way for it.

**Comment by ThuyNT13** [comment]
@mayureshjakhotia since you know L/MAMP could you integrate your PHP/BCRYPT with the UI setup yesterday here [index.html](https://github.com/CSims20/Eclipse-/blob/thuy/index.html)

**Comment by ThuyNT13** [comment]
we'll try to troubleshoot the Matrix Client-Server API.

**Comment by mayureshjakhotia** [comment]
Hey found one amazing tutorial. But it seems Matrix cannot work without setting up A domain name (e.g. from GoDaddy) and A virtual server running Debian 8 on a cloud service (AWS, DigitalOcean, etc.) or a physical server. (Also, you would need Synapse. The link has it all)
Here's the link for Matrix: https://blog.cryptoaustralia.org.au/2017/03/21/run-your-end-to-end-encrypted-chat-server-matrix-riot/
And I'll work on getting login integrated with UI.

**Comment by mayureshjakhotia** [comment]
I'm working on the UI Login (with CSS). Will update in some time.

**Comment by curiousjazz77** [comment]
Cool! We are still trying to get the matrix-js-sdk working. Someone here was helping us a bit. We think we'll have to use node.js

**Comment by mayureshjakhotia** [comment]
Yeah do check out the link I sent earlier. I think it has complete tutorial.

**Comment by mayureshjakhotia** [comment]
Also, @curiousjazz77 or @ThuyNT13 if possible you guys can set up MAMP or LAMP and test it in your environment. I'll be pushing code in a while.

**Comment by ThuyNT13** [comment]
@mayureshjakhotia is this how we would test it?[How to turn your home Ubuntu PC into a LAMP web server](https://www.howtogeek.com/howto/42480/how-to-turn-your-home-ubuntu-pc-into-a-lamp-web-server/)

**Comment by mayureshjakhotia** [comment]
Hi Guys, So I'm done with testing Login and Sign Up with UI. You can directly download it from github (https://github.com/CSims20/Eclipse-/tree/login-bcrypt/phpbcrypt) and use it. :)

![screen shot 2017-09-24 at 11 27 19 am](https://user-images.githubusercontent.com/17538538/30784486-8a952394-a11b-11e7-88f6-972583be1aa3.png)
![screen shot 2017-09-24 at 11 27 29 am](https://user-images.githubusercontent.com/17538538/30784484-8a947b42-a11b-11e7-8482-c6ec8474654c.png)
![screen shot 2017-09-24 at 11 27 42 am](https://user-images.githubusercontent.com/17538538/30784485-8a94e546-a11b-11e7-9f44-a6a28f82ad32.png)
![screen shot 2017-09-24 at 11 28 01 am](https://user-images.githubusercontent.com/17538538/30784487-8a95a170-a11b-11e7-99bb-a1cfcc1a7d05.png)

**Comment by mayureshjakhotia** [comment]
@ThuyNT13 Make sure you set password as 'root'. On github link, I have mentioned the steps too. Do let me know if anything else. @curiousjazz77 You can use MAMP if you want to. It's pretty straight forward. 

Also, I have included the name of logged in user on Home Screen left hand side (In the screenshot above, you can see my name "Mayuresh" below the avatar) :)

**Comment by ThuyNT13** [comment]
@mayureshjakhotia thanks! Jasmine had to take off so now it's just me. Yeah trying to figure out what I

**Comment by mayureshjakhotia** [comment]
@ThuyNT13 Oh okay I'm still available. Lemme know... :)

**Comment by ThuyNT13** [comment]
@mayureshjakhotia Jasmine had to take off so trying to figure out how to proceed. I'll test what you pushed up so that I could have something to present and go over what we tried and were able to implement and the issues. Thanks for all your help!

**Comment by mayureshjakhotia** [comment]
Hey it's a group thing. :)
I was also seeing Matrix but maybe need to set it up on AWS or somewhere. All the very best :D 
@ThuyNT13

**Comment by mayureshjakhotia** [comment]
If you aren't able to do it, lemme know once you got phpMyAdmin running (where you can use mysql).

**Comment by ThuyNT13** [comment]
@mayureshjakhotia Ming was saying that we should just use the AWS instances. Have you used that before?

**Comment by mayureshjakhotia** [comment]
Yes I have. But I no longer have an account there. I can tell you or even do it. It's very basic. Sign up at https://aws.amazon.com/ and then Say Instances and Get an Ec2 Instance. But I guess how much time is remaining to do that since setting up LAMP stack on it would require at least 30 min maybe.

**Comment by ThuyNT13** [comment]
yeah, i have an account but going to focus on figuring out to run your php file for now and ignoring AWS

**Comment by ThuyNT13** [comment]
@mayureshjakhotia am I running this command inside the PHP directory and is this actually starting the command that stars the server: 
```bash
sudo apt-get install lamp-server^
```

**Comment by mayureshjakhotia** [comment]
@ThuyNT13 That's just installing it.

**Comment by ThuyNT13** [comment]
so what starts the server

**Comment by mayureshjakhotia** [comment]
You can open up localhost and see if it works. The link says it too. Then go for phpmyadmin further in the tutorial and do sudo service apache2 restart which would restart apache and test phpmyadmin in browser wherein you can create database and table.

**Comment by ThuyNT13** [comment]
because there's this line which to me implies there was a start somewhere: 

`sudo service apache2 restart`

**Comment by mayureshjakhotia** [comment]
Isn't localhost running? I guess installation does it after installed then

**Comment by mayureshjakhotia** [comment]
After installation, does this work--> http://localhost/

**Comment by mayureshjakhotia** [comment]
@ThuyNT13 You don't need to do anything else there. Just install the server, check if localhost is running or run "sudo service apache2 restart"(or start), then go on to "sudo apt-get install phpmyadmin" directly. I guess should be fine.

**Comment by ThuyNT13** [comment]
how is it referencing/routing to the file as localhost brings up an Apache2 Ubuntu Default Page which confirms the setup went fine but says "You should replace this file (located at /var/www/html/index.html)".

**Comment by mayureshjakhotia** [comment]
Yup. So now go to location /var/www and put my folder containing all files there. Then go to localhost and then there would be an option with that folder name. Selecting that would get you to index page.

**Comment by ThuyNT13** [comment]
so I'm creating a directory within directory `/var/www` and transferring your php files there so it should read `index.php`?

**Comment by mayureshjakhotia** [comment]
Yes just put entire folder there containing all php, image and js files as it is. Don't change folder structure at all. Then just go to localhost and you should be seeing a link to that folder and clicking it would be all you need to do :)

**Comment by ThuyNT13** [comment]
so then the folder structure would be 
`Eclipse-/var/www/phpbcrypt`
with all the images, php and css files inside that?

**Comment by mayureshjakhotia** [comment]
No Eclipse~

**Comment by mayureshjakhotia** [comment]
Directly go into /var/www and paste it there.

**Comment by ThuyNT13** [comment]
so this is from the root directory?

**Comment by mayureshjakhotia** [comment]
Yes

**Comment by mayureshjakhotia** [comment]
Just like htdocs for MAMP users.

**Comment by ThuyNT13** [comment]
don't know ht docs.

**Comment by mayureshjakhotia** [comment]
Yes don't worry...Just do this and try localhost. Should be able to see UI.

**Comment by ThuyNT13** [comment]
i'll do that and get back to you if i have any problems

**Comment by mayureshjakhotia** [comment]
Sure :)

**Comment by mayureshjakhotia** [comment]
You need to set up database and table in phpMyAdmin after that as I have mentioned in ReadMe. Lemme know if any issues. :)

**Comment by ThuyNT13** [comment]
setup from `~/var/www/phpbcrypt` and ran `sudo service apache2 restart` but still only shows Apache2 Ubuntu Default Page so I'm going to try moving files out of phpbcrypt directly into `var/www/` to see if that works.

**Comment by mayureshjakhotia** [comment]
Go to localhost/foldername(phpbcrypt)

**Comment by mayureshjakhotia** [comment]
You don't need to restart

**Comment by mayureshjakhotia** [comment]
Something like this should work: http://localhost:8888/phpbcrypt/

**Comment by mayureshjakhotia** [comment]
Did it work?

**Comment by ThuyNT13** [comment]
no returned 404 error

**Comment by mayureshjakhotia** [comment]
That means it didn't find the file. Can you use Teamviewer? I can get access to your machine and use it.

**Comment by ThuyNT13** [comment]
don't have Teamviewer

**Comment by ThuyNT13** [comment]
give me a sec to download

**Comment by mayureshjakhotia** [comment]
Okay

**Comment by mayureshjakhotia** [comment]
Lemme know Partner ID once done.

**Comment by ThuyNT13** [comment]
my id 144 637 374

**Comment by mayureshjakhotia** [comment]
Password?

**Comment by ThuyNT13** [comment]
it9k61

**Comment by mayureshjakhotia** [comment]
Can you open terminal?

**Comment by mayureshjakhotia** [comment]
Can you enter password?

**Comment by ThuyNT13** [comment]
just mysql

**Comment by ThuyNT13** [comment]
that's okay

**Comment by ThuyNT13** [comment]
thanks! they've started demoing

**Comment by mayureshjakhotia** [comment]
All The Best @ThuyNT13 .....Hope the video helps :)

**Comment by mayureshjakhotia** [comment]
How was it? @ThuyNT13

**Comment by ThuyNT13** [comment]
@mayureshjakhotia just moved last week so only reading your question, now. Sorry! 

I wasn't the only one who presented w/o an actual functioning project (^_^). I winged the whole presentation as everybody else had a slide show. I was still trying to get things to work up until the last minute. And then I had to dash off because I needed to pack for my move to LA. All in all, it was a full weekend and it was fun to be cramming on new tech, freak out when things didn't work out and then just resign myself to "it is what it is".
